### DISCUSSION TITLE
Action Buttons in custom page

### DISCUSSION THREAD

**Discussion by anishsarkar** [filamentphp/filament#1390] [Help] [answered]
I have a custom page in which I have added a action button:

```

protected function getActions(): array
    {

        return [
            ButtonAction::make('query_status')
                // ->requiresConfirmation()
                ->label('Make update status query')
                ->hidden( !$this->canUserUpdatePaymentStatus() )
                ->icon('heroicon-s-status-online')
                ->action('updateStatus')
        ];
    
    }
```

I wanted to have confirmation modal, but when I tried to add  that, it throws an exception saying that the method is not available. 

The action **updateStatus** takes a couple of seconds, so I also wanted to disable it on **wire:loading**,  but I don't see how to do that. 

I also wanted to refresh the page, once the action is completed. Since this is livewire component, I am guessing that listeners would work the same way, as it does for livewire. 

Thanks in advance for any help.

**Comment by danharrin** [comment, marked as answer]
Hey, there's a bug report about the exception - #1370

**Reply by anishsarkar** [reply]
Thanks. I upgraded the packages. Now, it is working. 

It there way to pass the **wire:loading** to the button. I tried with extraAttributes. But it seems that the method is not available.

**Reply by danharrin** [reply]
Buttons in the newest versions get disabled automatically when they're clicked :)
